external/skiboot/libstb/secvar/crypto/crypto-openssl.c

```diff
@@ -253,39 +253,45 @@ int crypto_pkcs7_generate_w_signature(unsigned char **pkcs7, size_t *pkcs7Size,
 	//for every key pair get the data and add the signer to the pkcs7
 	for (int i = 0; i < keyPairs; i++) {
 		// get data of private keys
-		keyPEM = (unsigned char *)getDataFromFile(keyFiles[i],
-							  &keySizePEM);
-		if (!keyPEM) {
-			prlog(PR_ERR,
-			      "ERROR: failed to get data from priv key file %s\n",
-			      keyFiles[i]);
+		fp = fopen(keyFiles[i], "r");
+		if (fp == NULL) {
+			prlog(PR_ERR, "ERROR: failed to open file %s: %s\n",
+			      keyFiles[i], strerror(errno));
 			rc = INVALID_FILE;
 			goto out;
 		}
-		rc = crypto_convert_pem_to_der(
-			keyPEM, keySizePEM, (unsigned char **)&key, &keySize);
-		if (rc) {
+		rc = PEM_read(fp, &unnecessary_name, &unnecessary_hdr, &key,
+			      (long int *)&keySize);
+		OPENSSL_free(unnecessary_name);
+		OPENSSL_free(unnecessary_hdr);
+		fclose(fp);
+		//returns 1 on success
+		if (rc != 1) {
 			prlog(PR_ERR,
-			      "Conversion for %s from PEM to DER failed\n",
+			      "ERROR: failed to get data from priv key file %s\n",
 			      keyFiles[i]);
+			rc = INVALID_FILE;
 			goto out;
 		}
 		//get data from crt
-		crtPEM = (unsigned char *)getDataFromFile(crtFiles[i],
-							  &crtSizePEM);
-		if (!keyPEM) {
-			prlog(PR_ERR,
-			      "ERROR: failed to get data from cert file %s\n",
-			      crtFiles[i]);
+		fp = fopen(crtFiles[i], "r");
+		if (fp == NULL) {
+			prlog(PR_ERR, "ERROR: failed to open file %s: %s\n",
+			      keyFiles[i], strerror(errno));
 			rc = INVALID_FILE;
 			goto out;
 		}
-		rc = crypto_convert_pem_to_der(
-			crtPEM, crtSizePEM, (unsigned char **)&crt, &crtSize);
-		if (rc) {
+		rc = PEM_read(fp, &unnecessary_name, &unnecessary_hdr, &crt,
+			      (long int *)&crtSize);
```

Comment:
Hmm, that's a slightly concerning cast. We should probably make crtSize (and keySize) a long and then cast them to the required type - possibly with a test or assert to make sure they're non-negative. Harder to be stuffed around with integer casts than pointer casts, IMO.